packages/config/src/api/integrations.ts

```diff
@@ -2,26 +2,22 @@ import fetch from 'node-fetch'
 
 import { TestOptions } from '../types/options.js'
 
-type AvailableIntegration = {
-  slug: string
-  hostSiteUrl: string
-  hasBuild?: boolean
-}
+type AvailableIntegration = { slug: string; hostSiteUrl: string; hasBuild?: boolean }
 
-type GetAvailableIntegrationsOpts = {
-  testOpts: TestOptions
-  offline: boolean
-}
+type GetAvailableIntegrationsOpts = { testOpts: TestOptions; offline: boolean; netlifyApiHost: string }
 
 export const getAvailableIntegrations = async function ({
   testOpts,
   offline,
+  netlifyApiHost,
 }: GetAvailableIntegrationsOpts): Promise<AvailableIntegration[]> {
   if (offline) {
     return []
   }
   const { host } = testOpts
-  const baseUrl = new URL(host ? `http://${host}/` : `https://api.netlifysdk.com/`)
+  const extensionApiBaseUrl =
+    netlifyApiHost === 'api.netlify.com' ? 'https://api.netlifysdk.com/' : `https://api-staging.netlifysdk.com/`
```

Comment:
I'm not familiar with how this is all set up currently, but this mapping seems a bit brittle.

1. Would it be a big lift to map all prod/staging/etc. hosts from a single 'mode' or 'env' rather than mapping one set from another?
2. The assumption that any host that doesn't precisely match `api.netlify.com` should quietly map to the staging SDK API host seems brittle as well. Could we explicitly map prod -> and staging -> staging and throw otherwise? (or at least log an error or something)

Reply:
100%, I want to try it this way first and then see if I can let buildbot/cli pass in the right url depending on the environment